From 2f6d863f698d9c21eaea037076dd585c97ee7b96 Mon Sep 17 00:00:00 2001
From: Jackie Han <jackie.han@gmail.com>
Date: Fri, 25 Sep 2026 11:14:13 +0800
Subject: [PATCH] Modify ipq9570-sbe1v1k.dts for memory and crypto settings

Updated device tree source for ipq9570-sbe1v1k.dts with new memory configurations and disabled crypto nodes.

Signed-off-by: Jackie Han <jackie.han@gmail.com>
---
 .../linux/qualcommbe/dts/ipq9570-sbe1v1k.dts  | 75 +++++++++++++++----
 1 file changed, 60 insertions(+), 15 deletions(-)

diff --git a/target/linux/qualcommbe/dts/ipq9570-sbe1v1k.dts b/target/linux/qualcommbe/dts/ipq9570-sbe1v1k.dts
index f39badf135a028..1abc8c8142b7d4 100644
--- a/target/linux/qualcommbe/dts/ipq9570-sbe1v1k.dts
+++ b/target/linux/qualcommbe/dts/ipq9570-sbe1v1k.dts
@@ -23,13 +23,25 @@
 
 	chosen {
 		stdout-path = "serial0:115200n8";
+		bootargs = "cma=256M";
 	};
 
 	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
 		tzapp:tzapp@49B00000 {
 			no-map;
 			reg = <0x0 0x49B00000 0x0 0x00600000>;
 		};
+
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0x0 0x10000000>; /* 256MB = 0x10000000 */
+			linux,cma-default;
+		};
 	};
 
 	regulator_fixed_3p3: s3300 {
@@ -171,26 +183,33 @@
 	status = "okay";
 };
 
+/*
+ * No upstream driver currently binds these QCE nodes. Leaving them enabled
+ * prevents the GCC clock controller from completing sync_state(), which in
+ * turn leaves cpufreq-dt stuck in deferred probe.
+ */
+&crypto {
+	status = "disabled";
+};
+
+&cryptobam {
+	status = "disabled";
+};
+
 &rpm_requests {
 	regulators {
 		compatible = "qcom,rpm-mp5496-regulators";
 
 		ipq9574_s1: s1 {
-		/*
-		* During kernel bootup, the SoC runs at 800MHz with 875mV set by the bootloaders.
-		* During regulator registration, kernel not knowing the initial voltage,
-		* considers it as zero and brings up the regulators with minimum supported voltage.
-		* Update the regulator-min-microvolt with SVS voltage of 725mV so that
-		* the regulators are brought up with 725mV which is sufficient for all the
-		* corner parts to operate at 800MHz
-		*/
-			regulator-min-microvolt = <725000>;
+			regulator-min-microvolt = <587500>;
 			regulator-max-microvolt = <1075000>;
+			regulator-always-on;
+			regulator-boot-on;
 		};
 
-		mp5496_l5: l5 {
-			regulator-min-microvolt = <1800000>;
-			regulator-max-microvolt = <1800000>;
+		ipq9574_s2: s2 {
+			regulator-min-microvolt = <700000>;
+			regulator-max-microvolt = <1000000>;
 			regulator-always-on;
 			regulator-boot-on;
 		};
@@ -208,6 +227,9 @@
 	pinctrl-names = "default";
 	status = "okay";
 
+	#address-cells = <1>;
+	#size-cells = <0>;
+
 	card@0 {
 		compatible = "mmc-card";
 		reg = <0>;
@@ -488,6 +510,23 @@
 		reset-gpios = <&tlmm 41 GPIO_ACTIVE_LOW>;
 		reset-assert-us = <40000>;
 		reset-deassert-us = <150000>;
+
+		leds {
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			led@0 {
+				reg = <0>;
+				color = <LED_COLOR_ID_AMBER>;
+				function = LED_FUNCTION_WAN;
+			};
+
+			led@1 {
+				reg = <1>;
+				color = <LED_COLOR_ID_GREEN>;
+				function = LED_FUNCTION_WAN;
+			};
+		};
 	};
 
 	ethernet-phy-package@16 {
@@ -497,6 +536,12 @@
 		reg = <16>;
 
 		qcom,package-mode = "qsgmii";
+		/*
+		 * Match the 300 mV PQSGMII TX amplitude programmed by the
+		 * stock SSDK. The generic 600 mV default causes receive CRC
+		 * errors on the QSGMII channel connected to PPE port 3.
+		 */
+		qcom,tx-drive-strength-milliwatt = <300>;
 
 		qca8075_18: ethernet-phy@18 {
 			compatible = "ethernet-phy-ieee802.3-c22";
@@ -617,7 +662,7 @@
 
 	pcie@0 {
 		device_type = "pci";
-		reg = <0x0 0x0 0x1 0x0 0x0>;
+		reg = <0x0 0x0 0x0 0x0 0x0>;
 		#address-cells = <3>;
 		#size-cells = <2>;
 		ranges;
@@ -667,9 +712,9 @@
 	wake-gpios = <&tlmm 33 GPIO_ACTIVE_LOW>;
 	status = "okay";
 
-	pcie@0,0 {
+	pcie@0 {
 		device_type = "pci";
-		reg = <0x0 0x0 0x2 0x0 0x0>;
+		reg = <0x0 0x0 0x0 0x0 0x0>;
 		#address-cells = <3>;
 		#size-cells = <2>;
 		ranges;
